flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1638,6 +1636,14 @@ private boolean transitionState(ExecutionState currentState, ExecutionState targ
 		}
 	}
 
+	private String getLocationInformation() {
+		if (assignedResource != null) {
+			return assignedResource.getTaskManagerLocation().toString();
+		} else {
+			return "not deployed";
```

Comment:
Maybe it makes sense to indicate clearer that `not deployed` is an internal state?

Right now, the log message reads like this:

> blocking operator (1/2) (6c5f76b0e3ad109caf8f02517d79df29_74c3ec8044355e5795d2737d7de2ae46_0_0) switched from SCHEDULED to FAILED on not deployed.


Maybe something like this would be easier to understand:

> blocking operator (1/2) (6c5f76b0e3ad109caf8f02517d79df29_74c3ec8044355e5795d2737d7de2ae46_0_0) switched from SCHEDULED to FAILED on [unassigned resource].

Reply:
Sounds like a good idea. I will update it.